src/test/java/bio/terra/service/iam/sam/SamIamTest.java

```diff
@@ -57,4 +135,274 @@ public void testExtractErrorMessageDeepNested() {
+                snapshotId.toString(),
+                policy.toString())
+            ).thenReturn(Maps.newHashMap("policygroup-" + policy + "@firecloud.org", Collections.emptyList()));
+        }
+
+        assertThat(samIam.createSnapshotResource(userReq, snapshotId, Collections.emptyList()))
+            .isEqualTo(syncedPolicies.stream().collect(Collectors.toMap(
+                p -> p,
+                p -> "policygroup-" + p + "@firecloud.org"
+            )));
+    }
+
+    @Test
+    public void testCreateProfile() throws InterruptedException, ApiException {
+        final UUID profileId = UUID.randomUUID();
+        // Note: in our case, policies have a 1:1 relationship with roles
+        final List<IamRole> allPolicies = Arrays.asList(
+            IamRole.ADMIN,
+            IamRole.OWNER,
+            IamRole.USER
+        );
+
+        for (IamRole policy : allPolicies) {
+            when(samGoogleApi.syncPolicy(
+                IamResourceType.DATASNAPSHOT.getSamResourceName(),
+                profileId.toString(),
+                policy.toString())
+            ).thenReturn(Maps.newHashMap("policygroup-" + policy + "@firecloud.org", Collections.emptyList()));
+        }
+
+        samIam.createProfileResource(userReq, profileId.toString());
+    }
+
+    @Test
+    public void testAddPolicy() throws InterruptedException, ApiException {
+        final UUID id = UUID.randomUUID();
+        final String userEmail = "a@a.com";
+        when(samResourceApi.getPolicy(
+            IamResourceType.SPEND_PROFILE.getSamResourceName(),
+            id.toString(),
+            IamRole.OWNER.toString())
+        ).thenReturn(new AccessPolicyMembership()
+            .memberEmails(Collections.singletonList(userEmail)));
+        final PolicyModel policyModel =
+            samIam.addPolicyMember(userReq, IamResourceType.SPEND_PROFILE, id, IamRole.OWNER.toString(), userEmail);
+        assertThat(policyModel)
+            .isEqualTo(new PolicyModel()
+                .name(IamRole.OWNER.toString())
+                .addMembersItem(userEmail)
+            );
+        verify(samResourceApi, times(1)).addUserToPolicy(
+            IamResourceType.SPEND_PROFILE.getSamResourceName(),
+            id.toString(),
+            IamRole.OWNER.toString(),
+            userEmail
+        );
+    }
+
+    @Test
+    public void testDeletePolicy() throws InterruptedException, ApiException {
+        final UUID id = UUID.randomUUID();
+        final String userEmail = "a@a.com";
+        when(samResourceApi.getPolicy(
+            IamResourceType.SPEND_PROFILE.getSamResourceName(),
+            id.toString(),
+            IamRole.OWNER.toString())
+        ).thenReturn(new AccessPolicyMembership()
+            .memberEmails(Collections.emptyList()));
+        final PolicyModel policyModel =
+            samIam.deletePolicyMember(userReq, IamResourceType.SPEND_PROFILE, id, IamRole.OWNER.toString(), userEmail);
```

Comment:
Why don't you also need to mock samApi.removeUserFromPolicy() before the deletePolicyMember call?

Reply:
Nevermind - b/c the removeUserFromPolicy doesn't return anything, it just changes things on the SAM side. So, we can just mock what the policy would look like if the user was removed.